### ISSUE TITLE
Improve statsd metrics

### ISSUE THREAD

**Issue by schnie** [astronomerio/incubator-airflow#29]
We need to improve the metrics that airflow is reporting out via statsd. Let's get a quick PR in to Airflow that adds counters for `success` and `failed` `dag_runs`, tracking at least `dag_id` and `execution_date`. We should take a pass through the existing metrics and make sure they all work the way they should. I think at least one of the metrics may be a little off.

**Comment by schnie** [comment]
While we're using celery it might be nice to have an `airflow_worker_heartbeat` similar to `airflow_scheduler_heartbeat`.

**Comment by tedmiston** [comment]
We tested this with 0 DAGs, 1 DAGs, and many DAGs today.  We're seeing a temporary off-by-one error in DAG bag count size some of the time that appears to resolve itself over time, but haven't quite figured out how to reproduce it consistently yet.  It doesn't seem to always happen depending on the size of the DAG bag.

**Comment by schnie** [comment]
Any idea how multiple airflows affect the count?

**Comment by tedmiston** [comment]
Haven't tried that yet but we can test it.

**Comment by schnie** [comment]
@rhwang10 how is it calculating dagbag size? If I recall, something may have broken when they switched the way the scheduler processed the dag files. They spawn a new process for each file now right?

**Comment by rhwang10** [comment]
Yep, it gets the len(self.dags), which is a dictionary of {dag_id : dag object}, but I think since they started using the multiprocessing library, that dictionary will never contain more than one DAG

**Comment by schnie** [comment]
@rhwang10 how are you guys running this setup now? Were you able to get a locally editable version of airflow pointing to the metrics stack?

**Comment by rhwang10** [comment]
It is a locally editable version of airflow- in the docker-compose, we added a line each to the volumes of the worker, webserver, and scheduler to point to incubator-airflow

**Comment by tedmiston** [comment]
We weren't able to figure out why the non-dockerized local virtual env Airflow and the dockerized metrics stack couldn't talk to each other, but also didn't want to burn anymore time on troubleshooting that.

I was able to get running local incubator-airflow mounted into the relevant containers in the airflow-enterprise docker compose example so we can make changes quickly and just `docker-compose up / down` (basically like `pip install -e`).  Then we set that up on Rob's machine too, and it's working well for us so far with a fast dev cycle.

https://github.com/astronomerio/astronomer/compare/docker-volume-incubator-airflow

**Comment by rhwang10** [comment]
dag_bag size appears to be accurate if we query the database using the connection opened by the heartbeat method, filtering by whether the dag is active or not

**Comment by rhwang10** [comment]
update: trying to run multiple EE locally on different ports, but running into some issues with postgres when trying to spin both airflows up

**Comment by rhwang10** [comment]
Shifted priority from specific statsD metrics to kube-state metrics, but improving monitoring is still under active dev
